src/main/java/com/fleotadezuta/youthprogrammanager/facade/TicketChildTicketTypeFacade.java

```diff
@@ -0,0 +1,92 @@
+package com.fleotadezuta.youthprogrammanager.facade;
+
+import com.fleotadezuta.youthprogrammanager.mapper.ChildMapper;
+import com.fleotadezuta.youthprogrammanager.mapper.TicketMapper;
+import com.fleotadezuta.youthprogrammanager.mapper.TicketTypeMapper;
+import com.fleotadezuta.youthprogrammanager.model.*;
+import com.fleotadezuta.youthprogrammanager.persistence.document.TicketDocument;
+import com.fleotadezuta.youthprogrammanager.service.ChildService;
+import com.fleotadezuta.youthprogrammanager.service.TicketService;
+import com.fleotadezuta.youthprogrammanager.service.TicketTypeService;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.util.function.Tuple2;
+
+@Service
+@AllArgsConstructor
+@Slf4j
+public class TicketChildTicketTypeFacade {
+
+    private final TicketService ticketService;
+    private final TicketMapper ticketMapper;
+    private final TicketTypeService ticketTypeService;
+    private final TicketTypeMapper ticketTypeMapper;
+    private final ChildService childService;
+    private final ChildMapper childMapper;
+
+    public Flux<TicketTypeDto> getPotentialTicketTypes(String name) {
+        return ticketTypeService.findByName(name);
+    }
+    
+    private Mono<Tuple2<ChildDto, TicketTypeDto>> getChildAndTicketType(String childId, String ticketTypeId) {
+        return Mono.zip(
+                childService.findById(childId).map(childMapper::fromChildDocumentToChildDto),
+                ticketTypeService.findById(ticketTypeId).map(ticketTypeMapper::fromTicketTypeDocumentToTicketTypeDto)
+        );
+    }
+
+    private Mono<TicketDto> mapToTicketDto(TicketDocument ticketDocument, ChildDto child, TicketTypeDto ticketType) {
+        return Mono.just(ticketMapper.fromTicketDocumentToTicketDto(ticketDocument, child, ticketType));
+    }
```

Comment:
Úgy látom, hogy ez a két private metódus jellemzően párban van használva. Ez alapján össze is lehetne vonni egy metódussá, ami a `ticketDocument`-et kapja paraméterként és a végén egy `Mono<TicketDto>`-t ad vissza.

Ezzel a felhasználás is egyszerűbbé válna a mostaninál.